src/server/services/BruincastServices.js

```diff
@@ -166,6 +167,15 @@ class BruincastServices {
         delete entry._id;
         delete entry.timestamp;
       }
+
+      // Sort each course's media listings by date. If date is the same, then sort by title.
+      courseMedia.listings.sort(function(listing1, listing2) {
+        if (listing1.date < listing2.date) return -1;
```

Comment:
@chern This cannot be sorted on the database query?

Reply:
@rlorenzo Good point, hadn't thought of that; the latest commit now sorts the listings during db aggregation